lib/response-handler.js

```diff
@@ -114,18 +116,20 @@ module.exports = (bluebird, error, logger, parseLinkHeader, URI) => {
             // If we received more than one link header item, push them
             // all onto an array. If we received just one, push just that
             // one onto the array.
-            if (Array.isArray(links.item)) {
-                links.item.forEach((val) => {
-                    urls.push(val.url);
-                });
-            } else {
-                urls.push(links.item.url);
+            if (!Array.isArray(links.item)) {
+                urls = [links.item];
             }
 
             logger.debug(`Parsed out links: ${JSON.stringify(urls, null, 4)}`);
         }
 
-        return urls;
+        urls.forEach((url) => {
+            // Grabs all characters after artifact in the url.
+            url = url.substr(url.indexOf("artifact") + 8);
+            resolvedLinkHeaders.push(resolveLink(response, url));
```

Comment:
`resolveLink` doesn't take a response, it takes a url. A url and a link.

In fact, everything named `url` in this function probably should not be. They will be a `linkPath`. The url passed to `resolveLink` will be an actual url and it should come from the response.

Did you test this or am I going crazy?